code/world.dm

```diff
@@ -754,20 +400,39 @@ var/world_topic_spam_protect_time = world.timeofday
 
 #undef FAILED_DB_CONNECTION_CUTOFF
 
-/world/proc/do_topic_spam_protection(var/addr, var/key)
-	if (!config.comms_password || config.comms_password == "")
-		return "No comms password configured, aborting."
-
-	if (key == config.comms_password)
-		return 0
+/world/proc/do_auth_check(var/addr, var/auth, var/function)
+	//Check if rate limited
+	if(world_api_rate_limit[addr] != null && config.api_rate_limit_whitelist[addr] == null) //Check if the ip is in the rate limiting list and not in the whitelist
+		if(abs(world_api_rate_limit[addr] - world.time) < config.api_rate_limit) //Check the last request time of the ip
+			world_api_rate_limit[addr] = world.time // Set the time of the last request
+			log_debug("API - Rate Limit - Rate Limit Exceeded")
+			return 2 //Throttled
+
+	world_api_rate_limit[addr] = world.time // Set the time of the last request
+

```

Comment:
Missing piece of code here. A DB connection is not always up or even enabled, so it is a practice to check if we have it enabled before proceeding with the creation of a query. Otherwise you will end up with null pointer errors too. This will mean that the world/Topic system doesn't work without a valid DB, but, our code isn't meant to _run_ full feature without one, so that's fine.

The code you need:
```
	if (!establish_db_connection(dbcon))
		return 1 // Failed auth
```